src/modules/product/product.service.js

```
@@ -580,6 +574,132 @@
+ */
+export const deleteProductVariationService = async (variationId) => {
+  if (!isValidMongoId(variationId)) {
+    throw new Error("Invalid variation ID");
+  }
+
+  const variation = await ProductVariation.findByIdAndDelete(variationId);
+  if (!variation) {
+    throw new Error("Variation not found");
+  }
+
+  // Check if this was the last variation for the product
```

Biome (1.9.4)

[error] The catch clause that only rethrows the original error is useless.

An unnecessary catch clause can be confusing.
Unsafe fix: Remove the try/catch clause.


(lint/complexity/noUselessCatch)